components/sup/src/manager/service/spec.rs

```diff
@@ -266,6 +266,113 @@ impl ServiceSpec {
         }
         Ok(self)
     }
+
+    /// Given an `old` and a `new` spec, figure out what operations
+    /// are needed in order to turn the `old` state into the `new`
+    /// state.
+    ///
+    /// Here, `old` represents what a currently running service, while
+    /// `new` represents a new version of that spec that we wish to
+    /// make the currently running service. Both are `Option`s, in
+    /// order to capture the scenario in which, say, nothing is
+    /// currently running, but we wish to start a service, or where we
+    /// are running a service, but wish to stop it.
+    ///
+    /// Currently, it is *assumed* that both specs (when present)
+    /// refer to the same service.
+    ///
+    /// Returning `None` indicates that no operation is required.
+    pub(crate) fn reconcile(old: Option<ServiceSpec>,
+                            new: Option<ServiceSpec>)
+                            -> Option<ServiceOperation> {
+        // TODO (CM): Invariant:
```

Comment:
Do you still want this here?

Reply:
No, it's not necessary, particularly with the upcoming PR I have to actually make the configuration dynamic. The way this function is called, the spec identifiers should be the same, but if they aren't, we'll still do the right thing (i.e., restart).

This was more a note to myself while working through the logic.